### PR TITLE
Remove some deprecated functions

### PR DESCRIPTION
**Pull request by photodude**
- JArrayHelper has been deprecated use Joomla\Utilities\ArrayHelper;
- ~~JString has been depreciated use use Joomla\String\StringHelper;~~ Use JString for now and StringHelper in 3.5 dev
- JError has been depreciated use JFactory::getApplication()->enqueueMessage()

### CODE REVIEW

File: libraries/legacy/model/admin.php [hunk 1]
```diff
@@ -834,8 +836,8 @@ protected function generateNewTitle($category_id, $alias, $title)
 
 		while ($table->load(array('alias' => $alias, 'catid' => $category_id)))
 		{
-			$title = JString::increment($title);
-			$alias = JString::increment($alias, 'dash');
+			$title = String::increment($title);
+			$alias = String::increment($alias, 'dash');
```
Comment (asika32764) [inline libraries/legacy/model/admin.php]: I think we should use `StringHelper` since PHP7 will not allow type as class name.

Comment (mbabker) [inline libraries/legacy/model/admin.php]: StringHelper isn't in the `staging` branch, only `3.5-dev`.

Comment (photodude) [inline libraries/legacy/model/admin.php]: So would it be better to continue to use `JString` in the current staging branch and avoid `Joomla\String\String` due to the PHP7 type as class name issue?

(on the other hand I believe JString is just extending or aliasing `Joomla\String\String` so I'm not sure using JString would really avoid the PHP7 type as class name issue)

Comment (mbabker) [inline libraries/legacy/model/admin.php]: For the `staging` branch I'd stick with `JString`.  All the PHP 7 fixes are in the `3.5-dev` branch anyway so getting 3.4 working on PHP 7 is out of the question.

Comment (photodude) [inline libraries/legacy/model/admin.php]: Sounds good, I've reverted the file to use `JString` as recommended for the current staging branch.

**Comment by photodude** [comment]
@mbabker The suggested change to continue to use JString has been implemented.

**Comment by photodude** [comment]
Closing in favor of existing changes in pending release of J3.5

**Pull request closed without merging**